app/lib/external_id_provider.rb

```diff
@@ -0,0 +1,5 @@
+class ExternalIdProvider
+  def self.generate_id
+    SecureRandom.base58(8)
+  end
```

Comment:
Seeing as the only place we're using this code in this app currently we're immediately downcasing the output, I'd suggest either changing `generate_id` to also call `downcase`.

If we're keen to preserve compatibility with the code in forms-admin we could instead add a new method `generate_case_insensitive_id` and call that instead.

The current code has the behaviour split between the definition and the callsite, which I found confusing on first reading.

Reply:
Happy to change this, I think I also prefer it all living in one place, and the model can trust that the ID generator is giving it something appropriate.